RecoverVm/AttachOsDiskAsDataDiskToRecoveryVm.ps1

```diff
@@ -0,0 +1,63 @@
+﻿function AttachOsDiskAsDataDiskToRecoveryVm   (
+    [string] $ServiceName , 
+    [string] $VName ,    
+    [string] $RecoveryAdmin = 'recoveryAdmin',
+    [string] $RecoveryPW = [guid]::NewGuid().ToString()
+    )
+{
+    $RecoveryVMName = 'RC' + $(get-date -f yyMMddHHmm)    
+
+    #get defect vm object and export cfg to disk (temp)
+    $vm = Get-AzureVM -ServiceName $ServiceName -Name $VMName
+
+    if ( ! $vm ) 
+    {
+        return
+    }
+
+    #https://github.com/Azure/azure-powershell/pull/2050 bug
+    #export vm cfg to temp dir for later exact recreation (disk,net settings)
+    $VMExportPath = $env:TEMP + '\' + $vm.VM.OSVirtualHardDisk.DiskName + '.xml'
+
+    $vm | Export-AzureVM -Path $VMExportPath
+    Write-Host 'Original VM Configuration written to ' $VMExportPath -ForegroundColor Yellow
+
+    Write-host 'Creating VM (may take a few minutes)'
+    
+    #get storage account name from current os disk url
+    $SAName = $vm.VM.OSVirtualHardDisk.MediaLink.Authority.Split(".")[0]   
+
+    #set current sa to be os disk sa -> will create reco vhd on same sa
+    $Sub = Get-AzureSubscription -Current 
+    Set-AzureSubscription -CurrentStorageAccountName $SAName -SubscriptionId $sub.SubscriptionId
+    
+    if ( $vm.VM.OSVirtualHardDisk.OS -eq 'Windows' )
+    {
+        $RecoveryImage =  Get-AzureVMImage |where ImageFamily -eq 'Windows Server 2012 R2 Datacenter'  | sort PublishedDate -Descending | select  -First 1
+        $recoveryVM = New-AzureQuickVM -Windows -WaitForBoot -ServiceName $ServiceName -Name $RecoveryVMName -InstanceSize $vm.InstanceSize -AdminUsername $RecoveryAdmin -Password $RecoveryPW -ImageName $RecoveryImage.ImageName -EnableWinRMHttp
+    }
+    else
+    {
+        $RecoveryImage =  Get-AzureVMImage |where ImageFamily -eq 'Ubuntu Server 14.04 LTS'  | sort PublishedDate -Descending | select  -First 1
```

Comment:
Note: this is 14.04 for RDFE whereas the CRP version uses 16.04.  Perhaps this is intentional, but we should validate.

Reply:
Will submit a separate PR request to allow the rdef script to take the Version info as a parameter and support others Centos/Suse/Oracle/Redhat as well.